File: Modules/ServerTypes/Administration/WindowsGroupColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { WindowsGroupRow } from "./WindowsGroupRow";

export interface WindowsGroupColumns {
    Id: Column<WindowsGroupRow>;
    GroupName: Column<WindowsGroupRow>;
}

export class WindowsGroupColumns extends ColumnsBase<WindowsGroupRow> {
    static readonly columnsKey = 'Administration.WindowsGroup';
    static readonly Fields = fieldsProxy<WindowsGroupColumns>();
}

File: Modules/Administration/WindowsGroup/WindowsGroupColumns.cs
using Serenity.ComponentModel;
using System.ComponentModel;

namespace WinAuthSample.Administration.Columns
{
    [ColumnsScript("Administration.WindowsGroup")]
    [BasedOnRow(typeof(WindowsGroupRow), CheckNames = true)]
    public class WindowsGroupColumns
    {
        [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
        public int Id { get; set; }
        [EditLink]
        public string GroupName { get; set; }
    }
}